raiden/tests/integration/test_integration_events.py

```diff
@@ -525,3 +525,94 @@ def test_secret_revealed(raiden_chain, deposit, settle_timeout, token_addresses)
         app0, deposit - amount, [],
         app1, deposit + amount, [],
     )
+
+
+@pytest.mark.parametrize('number_of_nodes', [2])
+def test_clear_closed_queue(raiden_network, token_addresses, deposit):
+    """ Closing a channel clears the respective message queue. """
+    app0, app1 = raiden_network
+    registry_address = app0.raiden.default_registry.address
+    token_address = token_addresses[0]
+    chain_state0 = views.state_from_app(app0)
+    token_network_identifier = views.get_token_network_identifier_by_token_address(
+        chain_state0,
+        app0.raiden.default_registry.address,
+        token_address,
+    )
+    token_network = views.get_token_network_by_identifier(
+        chain_state0,
+        token_network_identifier,
+    )
+
+    channel_identifier = get_channelstate(app0, app1, token_network_identifier).identifier
+
+    assert channel_identifier in token_network.partneraddresses_to_channels[
+        app1.raiden.address
+    ]
+
+    app1.raiden.transport.stop()
+    app1.raiden.transport.get()
+
+    # make a direct transfer to ensure the nodes have communicated
+    amount = 10
+    payment_identifier = 1337
+    app0.raiden.direct_transfer_async(
+        token_network_identifier,
+        amount,
+        app1.raiden.address,
+        identifier=payment_identifier,
+    )
+
+    # assert the specific queue is present
+    chain_state0 = views.state_from_app(app0)
+    queues0 = views.get_all_messagequeues(chain_state=chain_state0)
+    assert [
+        (queue_id, queue)
+        for queue_id, queue in queues0.items()
+        if queue_id.recipient == app1.raiden.address and
+        queue_id.channel_identifier == channel_identifier and queue
+    ]
+
+    app1.raiden.transport.start(app1.raiden)
```

Comment:
Because you're starting the app1 again, the protocol will proceed and the queue will be cleared once the transfer is done ([code ref](https://github.com/raiden-network/raiden/blob/8b23b7242e9b2d24e5e52526ee389e2f697d9510/raiden/transfer/node.py#L494)).

If you want to test that the queue is cleared *because* of the channel close, then you must have pending transfers in the queue, which we can enforce either with a custom event handler or by never starting the transport again.

Reply:
Oh, you're right. I'm addressing this then by not restarting the transport.